db/version_set.cc

```diff
@@ -872,10 +872,17 @@ Status VersionSet::Recover(bool* save_manifest) {
   if (!s.ok()) {
     return s;
   }
-  if (current.empty() || current[current.size() - 1] != '\n') {
+  const size_t size = current.size();
+  if (size == 0 || (current[size - 1] != '\n' && current[size - 1] != '\r')) {
```

Comment:
Change looks fine as is but I wonder when do we have this condition (file ending with \r only not with the combination \r\n) ... wasn't this condition a side effect of not having the change below? (as in the first time we call recover it goes and clears the \n but not the \r and now the file is all weird).

Reply:
Doesn't seem like the `current.resize(current.size() - 1);` causes this problem since `current` is just a string that gets reinitialized each time `Recover` is called. Removing the /n in `current` (for posix devices) on the first run shouldn't cause the next value of `current` after `Recover` is called again to be messed up since it's reinitialized with `ReadFileToString`. Seems like value of `current` is only adjusted for setting `dscname`. Not why file would be ending with \r, but this fix seems to have fixed whatever was causing it, but the original author of this commit was unable to remember the specifics.